lib/deployTask.ts

```diff
@@ -0,0 +1,61 @@
+import { task, types } from "hardhat/config";
+const privateKey = process.env.DEPLOYER_PRIVATE_KEY;
+
+task("stakedCelo:deploy", "Deploys contracts with custom hardhat config options.")
+  .addParam(
+    "overrideNetwork",
```

Comment:
this param is quite hacky imo, ideally is should be able to read the --network that was passed to the deploy scripts, but it was not found in the taskArgs that was passed to the setAction method - quite odd, so i am having introduces this flag.

Reply:
Could you get the network from `hre.network` instead of `taskArgs` since `--network` is a hardhat global option instead of a taks arg?